app/src/main/java/com/vultisig/wallet/ui/models/ChainSelectionViewModel.kt

```diff
@@ -60,16 +62,31 @@ internal class ChainSelectionViewModel @Inject constructor(
                             vault.hexChainCode,
                             coin.coinType.derivationPath()
                         )
-                        val address = chainAccountAddressRepository.getAddress(
-                            coin.coinType,
-                            PublicKey(derivedPublicKey.hexToByteArray(), PublicKeyType.SECP256K1)
-                        )
-                        vault.copy(
-                            coins = vault.coins + coin.copy(
-                                address = address,
-                                hexPublicKey = derivedPublicKey
+                        if (coin.chain == Chain.mayaChain) {
+                            vault.copy(
+                                coins = vault.coins + coin.copy(
+                                    address = MayaChainHelper(
+                                        vault.pubKeyECDSA,
+                                        vault.hexChainCode
+                                    ).getCoin()?.address ?: "",
+                                    hexPublicKey = derivedPublicKey
+                                )
                             )
-                        )
+                        } else {
+                            val address = chainAccountAddressRepository.getAddress(
+                                coin.coinType,
+                                PublicKey(
+                                    derivedPublicKey.hexToByteArray(),
+                                    PublicKeyType.SECP256K1
+                                )
+                            )
+                            vault.copy(
+                                coins = vault.coins + coin.copy(
+                                    address = address,
+                                    hexPublicKey = derivedPublicKey
+                                )
+                            )
+                        }
```

Comment:
You can extend the `chainAccountAddressRepository::getAddress` method, and add MayaChain support there, so we can use one method for querying addresses everywhere (or at least a single entity) and avoid duplication of that code in TokenSelectionViewModel